src/loki/matrix_action.cc

```diff
@@ -131,7 +130,10 @@ namespace valhalla {
 
       //check the distances
       auto max_location_distance = std::numeric_limits<float>::min();
-      check_distance(sources, targets, max_distance.find("sources_to_targets")->second, max_location_distance);
+      auto matrix_max_dist = max_matrix_distance.find(costing);
+      if (matrix_max_dist == max_matrix_distance.end())
```

Comment:
Well my thinking was that if someone made a matrix query using a mode of travel like hov or truck then it would just default to auto. So maybe I should add a max_matrix_distance parameter for each of those modes as well?